packages/medusa/src/services/order.ts

```diff
@@ -459,27 +536,36 @@ class OrderService extends TransactionBaseService {
 
   /**
    * Creates an order from a cart
-   * @param cartId - id of the cart to create an order from
    * @return resolves to the creation result.
+   * @param cartOrId
    */
-  async createFromCart(cartId: string): Promise<Order | never> {
+  async createFromCart(cartOrId: string | Cart): Promise<Order | never> {
```

Comment:
question: why are we adding support for a full cart object as well?

Reply:
because sometimes we already have the cart in the hand and if the user knows what he is doing I think he should be able to pass it as we create from cart. I am fine to also go back to passing the cart id only but the idea was to allow to reduce the number of queries if some users (like I did) need to increase their performance. Also, it allows to not surcharge the db when we perform a big number of queries. wdyt?